Constellation mini docs (#214)
edgelesssys/constellation: docs/docs/getting-started/mini-constellation.md

# Mini Constellation

With `constellation mini`, users can quickly deploy and test Constellation without the need for a cloud subscription.

The command uses virtualization to create a local cluster with one control-plane and one worker node.

:::info

Since mini Constellation is running on your local system, please note that common cloud features, such as load-balancing,
attaching persistent storage, or auto-scaling, are unavailable.

:::

## Prerequisites

* [Constellation CLI](./install.md#install-the-constellation-cli)

> * Linux operating system on amd64 hardware

Reviewer: ```suggestion
* Linux operating system on amd64 
```
Or similar but the distro also needs to be amd64. Not sure if people actually run i686 distros though anymore, they would already fail running the pre-built CLI.

* At least 6GB RAM

> * A CPU with at least 4 cores

Reviewer: ```suggestion
* A x86-64 CPU with at least 4 cores
* Hardware virtualization enabled in the BIOS/UEFI (often referred to as Intel VT-x or AMD-V/SVM)
```
Unfortunately many HW vendors did disable virtualization in the BIOS/UEFI by default for many years so I would point that out.

Maybe I would also move this to the top of the list.

* 20GB of free disk space
* [KVM kernel module](https://www.linux-kvm.org/page/Main_Page) enabled

>   * Ensure your user is allowed to access `/dev/kvm` by running the following
> 
>     ```shell
>     sudo chmod 666 /dev/kvm
>     ```
> 

Reviewer: ```suggestion
```
Should hopefully be solved with https://github.com/edgelesssys/constellation/pull/198/commits/f3957c41b5cc0e987634d86823334d2fd481055e :)

* [Docker](https://docs.docker.com/engine/install/)

> * [xsltproc](http://xmlsoft.org/xslt/xsltproc.html)

Reviewer: Is there an install guide available for this on the internet? Would link to that instead of the man page if there's a good one out there for common distros.

# Mini Constellation

> With `constellation mini`, users can deploy and test Constellation without the need for a cloud subscription.
> 

Reviewer 2: ```suggestion
With `constellation mini`, users can deploy and test Constellation locally without the need for a cloud subscription.

```

attaching persistent storage, or autoscaling, are unavailable.

:::

## Prerequisites

* [Constellation CLI](./install.md#install-the-constellation-cli)
* A x86-64 CPU with at least 4 cores
  * Recommended are 6 cores or more
* Hardware virtualization enabled in the BIOS/UEFI (often referred to as Intel VT-x or AMD-V/SVM)
* At least 4 GB RAM
  * Recommend are 6 GB or more
* 20 GB of free disk space
* a Linux operating system
* [KVM kernel module](https://www.linux-kvm.org/page/Main_Page) enabled
* [Docker](https://docs.docker.com/engine/install/)
* [xsltproc](https://gitlab.gnome.org/GNOME/libxslt/-/wikis/home)
  * Install on Ubuntu:

    ```bash
    sudo apt install xsltproc
    ```

  * Install on Fedora

    ```bash
    sudo dnf install xsltproc
    ```

* (Optional) [`virsh`](https://www.libvirt.org/manpages/virsh.html) to observe and access your nodes

## Create your cluster

Setting up your mini Constellation cluster is as easy as running the following command:

```bash
constellation mini up
```

This will configure your current directory as the working directory for Constellation.

> All `constellation` commands concerning this cluster should be issued from this directory.

Reviewer 2: ```suggestion
All `constellation` commands concerning this cluster need to be issued from this directory.
```
